libraries/AP_RangeFinder/AP_RangeFinder_BLPing.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "RangeFinder.h"
+#include "RangeFinder_Backend.h"
+
+class AP_RangeFinder_BLPing : public AP_RangeFinder_Backend
+{
+
+public:
+
+    // constructor
+    AP_RangeFinder_BLPing(RangeFinder::RangeFinder_State &_state,
+                            AP_RangeFinder_Params &_params,
+                            AP_SerialManager &serial_manager,
+                            uint8_t serial_instance);
+
+    // static detection function
+    static bool detect(AP_SerialManager &serial_manager, uint8_t serial_instance);
```

Comment:
This is the pattern, of course, but we now have a singleton for serialmanager.

Reply:
I might leave it accepting the serial manager argument and let someone else blast them for all the drivers at once.